### DISCUSSION TITLE
How to start for Dummies

### DISCUSSION THREAD

**Discussion by NurdaRyspay** [ria-com/nomeroff-net#128] [Q&A] [answered]
Sorry I am an Android developer, so I don't know how to start, I have issues with library compatibility, pls help me
Tutorial didn't work with requirements below. Thank you

**Comment by ApelSYN** [comment, marked as answer]
Try to run the example in a docker container. If you encounter errors, write to us about them.

**Reply by eab64** [reply]
Добрый день, вопрос был от меня, спасибо за ответ!
P.s. Скорее всего вопросы будут глупо звучать, никогда не работал с open cv и tensorflow.
После того как скачал все по шагам, та же версия питона, те же версий библиотек, модель для теста вроде взял последний, отсюда - https://github.com/ria-com/nomeroff-net/tree/master/examples, у меня появляются ошибки по совместимости numpy и TF.
Но я возможно не правильно скачал натренированную модель, не совсем понятно что здесь -https://nomeroff.net.ua/models/ именно надо положить в директорию models.
Меня интересует модель для Казахстанских номеров, получается мне нужен файлик внутри MODEL/OCR/kz/ с расширением .h5 или скачать все что там есть mcnn. detectron и т.д?

**Reply by ApelSYN** [reply]
> P.s. Скорее всего вопросы будут глупо звучать, никогда не работал с open cv и tensorflow.

Кроме opencv и tensotflow мы используем еще библиотеку pytorch . Требования у версиям смотрите в файле [requirements.txt](https://github.com/ria-com/nomeroff-net/blob/master/requirements.txt)

> Но я возможно не правильно скачал натренированную модель

Nomeroff Net сам при первом запуске скачает нужные модели, Вам не нужно что-то дополнительно скачивать.

> Меня интересует модель для Казахстанских номеров

Я вам рекомендую запустить все в Docker-контейнере, там уже все версии правильно подобраны и все заработает без дополнительной настройки. https://github.com/ria-com/nomeroff-net/tree/master/docker

**Reply by eab64** [reply]
File "license_plate.py", line 33, in <module>
    nnet.loadModel(NOMEROFF_NET_DIR)
  File "/var/www/nomeroff-net/NomeroffNet/DetectronDetector.py", line 56, in loadModel
    from centermask.config import get_cfg
ModuleNotFoundError: No module named 'centermask'


Внутри докера когда запускаю скрипт(python3 license_plate.py), в классе Detector() метод loadModel падает пытаясь импортировать def get_cfg()
в IDE тоже подсвечивает что нет такого модуля.

Build делал CPU версию.
Командой внутри run.sh запускал контейнер.

В чем может быть проблема не подскажете?

**Reply by ApelSYN** [reply]
Предполагается что все примеры запускаются в папке **./examples**, можно запускать и в другой локации, но тогда нужно в примере поправить пути 
```python
# NomeroffNet path
NOMEROFF_NET_DIR = os.path.abspath('../')
sys.path.append(NOMEROFF_NET_DIR)
```
...
```python
# Detect numberplate
img_path = 'images/example2.jpeg'
img = mpimg.imread(img_path)
```

**Reply by eab64** [reply]
Все, заработало, большое спасибо за мгновенный фидбэк и удачи в развитий проекта)
